bigchaindb/elections/election.py

```diff
@@ -199,36 +204,43 @@ def has_concluded(self, bigchain, current_votes=[], height=None):
         election_pk = self.to_public_key(self.id)
         votes_committed = self.get_commited_votes(bigchain, election_pk)
         votes_current = self.count_votes(election_pk, current_votes)
-        current_validators = self.get_validators(bigchain, height)
 
-        if self.is_same_topology(current_validators, self.outputs):
-            total_votes = sum(current_validators.values())
-            if (votes_committed < (2/3) * total_votes) and \
-                    (votes_committed + votes_current >= (2/3)*total_votes):
-                return True
+        if self.has_validator_set_changed(bigchain):
+            return False
```

Comment:
It would be better to move this check to the beginning of the function so that the votes are not counted unnecessarily.